File: JavaScript/Tarea JS Clase 1/Teclado.js
import { DispositivoEntrada } from './DispositivoEntrada.js';


export class Teclado extends DispositivoEntrada {

    static contadorTeclado = 0;

    constructor(marca){
        super('Teclado', marca)
        this.idTeclado = ++Teclado.contadorTeclado;
    }

    toString() {  
        return `Teclado ID: ${this._idTeclado}, ${super.toString()}`;  
    }  

}

File: Java/Clase01/MundoPC/src/ar/com/system2025/mundopc/Teclado.java
package ar.com.system2025.mundopc;

public class Teclado extends DispositivoEntrada {

    private static int contadorTeclado = 0;
    private final int idTeclado;

    public Teclado(String tipoEntrada, String marca){
        super(tipoEntrada, marca);
        this.idTeclado = ++Teclado.contadorTeclado;
    }

    @Override
    public String toString() {
        return "\nTeclado:" +
                "\n idTeclado:" + idTeclado + super.toString();
    }

    public static void imprimirTeclado(Teclado teclado){
        System.out.println(teclado.toString());
    }

    public static void main(String[] args) {
        Teclado teclado1 = new Teclado("USB", "Genious");
        imprimirTeclado(teclado1);
    }

}
